app/src/main/java/com/infomaniak/drive/ui/fileList/FileListFragment.kt

```diff
@@ -268,6 +276,41 @@ open class FileListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
         getBackNavigationResult<Int>(REFRESH_FAVORITE_FILE) {
             mainViewModel.refreshActivities.value = true
         }
+
+        getBackNavigationResult<Int>(DownloadProgressDialog.OPEN_BOOKMARK) { fileId ->
+            FileController.getFileProxyById(fileId, customRealm = mainViewModel.realm)?.openBookmarkIntent()
+        }
+    }
+
+    private fun File.openBookmarkIntent() {
+        kotlin.runCatching {
+            if (name.endsWith(".url")) openUrlIntent(getUrlFromUrlFile(getUri(requireContext()).second))
+            else openUrlIntent(getUrlFromWebloc(getUri(requireContext()).second))
```

Comment:
```suggestion
            openUrlIntent(
                if (name.endsWith(".url")) getUrlFromUrlFile(getUri(requireContext()).second)
                else getUrlFromWebloc(getUri(requireContext()).second)
            )
```

Reply:
Uh ... No sorry but I don't accept this proposal, it is completely ugly, I will simplify it otherwise